src/NServiceBus.Core.Tests/Recoverability/SecondLevelRetries/SecondLevelRetriesTests.cs

```diff
@@ -16,126 +16,156 @@
     [TestFixture]
     public class SecondLevelRetriesTests
     {
+        FakeDispatchPipeline dispatchPipline;
+
+        [SetUp]
+        public void SetUp()
+        {
+            dispatchPipline = new FakeDispatchPipeline();
+        }
+
         [Test]
         public async Task ShouldRetryIfPolicyReturnsADelay()
         {
-            var delay = TimeSpan.FromSeconds(5);
-            var fakeDispatchPipeline = new FakeDispatchPipeline();
-            var behavior = new SecondLevelRetriesBehavior(new FakePolicy(delay), "test-address-for-this-pipeline");
+            var message = CreateMessage(id: "message-id", slrRetryHeader: "1");
+            var context = CreateContext(message);
 
-            var context = CreateContext("someid", 1, fakeDispatchPipeline);
+            var delay = TimeSpan.FromSeconds(5);
+            var behavior = new SecondLevelRetriesBehavior(new FakePolicy(delay), "deferral-address", GetFailureInfoStorage());
 
-            await behavior.Invoke(context, () => { throw new Exception("testex"); });
+            await behavior.Invoke(context, () => { throw new Exception("exception-message"); });
+            await behavior.Invoke(context, () => Task.FromResult(0));
```

Comment:
I'm on the fence about these stateful tests. Wouldn't it be easier to verify/setup on single Invoke calls?